### DISCUSSION TITLE
AWS Lambda, Secrets Manager and .pem file

### DISCUSSION THREAD

**Discussion by SanveerBenipal** [oracle/python-oracledb#329] [Q&A] [answered]
Hi Everyone,

I need to connect to the DB, using thin-client. so I've stored the .pem file in secrets manager,  in lambda I can create file in /tmp folder, is there way to pass the variable directly.
Thanks

**Comment by sanveers** [comment]
ewallet.pem and tnsnames.ora is stored in secrets manager --> lamba fetch the secret --> create dir '/tmp/config_dir/', then creating two files in it ewallet.pem and tnsnames.ora 

issue was when I stored wallet credential in secrets manager.  it removed the Line feed,
so connection was not establishing because of missing character codes.

so I converted .pem to base64 then stored in secret manager
so after that when I created the file it worked from base64 and stored in tmp location of lambda it worked.

**Reply by anthony-tuininga** [reply, marked as answer]
Python requires that the .pem file be on disk for the regular ssl module to work. There are other ways but these are not part of the standard library which python-oracledb uses for its TLS support.
